Update Readme 24.2 (#3)
DevExpress-Examples/xaf-how-to-sort-a-listview-in-code: Readme.md

<!-- default badges list -->
[![](https://img.shields.io/badge/Open_in_DevExpress_Support_Center-FF7200?style=flat-square&logo=DevExpress&logoColor=white)](https://supportcenter.devexpress.com/ticket/details/E1276)
[![](https://img.shields.io/badge/📖_How_to_use_DevExpress_Examples-e9f6fc?style=flat-square)](https://docs.devexpress.com/GeneralInformation/403183)
[![](https://img.shields.io/badge/💬_Leave_Feedback-feecdd?style=flat-square)](#does-this-example-address-your-development-requirementsobjectives)
<!-- default badges end -->

# XAF - How to sort a ListView in code

This example sorts list view data by a class property and prevents users from modifying the sorting settings.

![Data sorted by Modified On column value](blazor-sorted-grid.png)

## Implementation Details

> 1. Create a view controller in the application model. Configure sorting settings for the list view's column.

Reviewer: ```suggestion
1. Create a view controller in the application model and configure sorting settings for the list view's column:
```

    _File to review:_ [SortListViewController.cs](CS/EFCore/SortListViewEF/SortListViewEF.Module/Controllers/SortListViewController.cs) 
    ```cs
    protected override void OnActivated() {
        base.OnActivated();
        string propertyName = nameof(Issue.ModifiedOn);
        bool demoFlag = true;
        // This code applies a client side sorting.
        if(demoFlag) {
            IModelColumn columnInfo = View.Model.Columns[propertyName];
            if(columnInfo != null) {
                columnInfo.SortIndex = 0;
                columnInfo.SortOrder = ColumnSortOrder.Descending;
            }
        } else {
            // This code is used for the server side sorting.
            if(View.Model.Sorting[propertyName] == null) {
                IModelSortProperty sortProperty = View.Model.Sorting.AddNode<IModelSortProperty>(propertyName);
                sortProperty.Direction = SortingDirection.Descending;
                sortProperty.PropertyName = propertyName;
            }
        }
    }
    ```

> 2. Implement platform-dependent controllers that disable the sorting functionality in underlying grid controls. 

Reviewer: ```suggestion
2. Implement platform-dependent controllers that disable the sorting functionality in underlying grid controls:
```

    _File to review:_ [BlazorSortRootListViewController.cs](CS/EFCore/SortListViewEF/SortListViewEF.Blazor.Server/Controllers/BlazorSortRootListViewController.cs) 
    ```cs
    protected override void OnViewControlsCreated() {
        base.OnViewControlsCreated();
        if(View.Editor is DxGridListEditor gridListEditor) {
            foreach(DxGridDataColumnModel columnModel in gridListEditor.GridDataColumnModels) {
                columnModel.AllowSort = false;
            }
        }
    }
    ```

    _File to review:_ [WinSortListViewController.cs](CS/EFCore/SortListViewEF/SortListViewEF.Win/Controllers/WinSortListViewController.cs) 

    ```cs
    protected override void OnViewControlsCreated() {
        base.OnViewControlsCreated();
        if(View.Editor is GridListEditor gridListEditor) {
            gridListEditor.GridView.OptionsCustomization.AllowSort = false;
        }
    }
    ```

> This approach allows you to sort both nested and root list views, and will work if the server mode is enabled in the list view.

Reviewer: ```suggestion
This approach allows you to sort both nested and root list views, and works if server mode is enabled in the list view.
```
